sagemaker-neo-notebooks/edge/cpp-integration/tutorial.cc

```diff
@@ -0,0 +1,622 @@
+  } else {
+    auto &model_file = get_object_outcome.GetResultWithOwnership().GetBody();
+
+    // download the sample file
+    const char *output_filename = compiled_filename.c_str();
+    std::ofstream output_file(output_filename, std::ios::binary);
+    output_file << model_file.rdbuf();
+  }
+}
+
+void DownloadNpyData() {
+  Aws::S3::S3Client s3_client = getS3Client();
+
+  // this is a preprocess image file in our sample directory
+  // this can be changed to any s3 bucket you specified
+  Aws::String aws_bucket_name = "neo-ai-dlr-test-artifacts";
+  Aws::String aws_object_name = "test-data/dog.npy";
+  std::string filename = "dog.npy";
+
+  Aws::S3::Model::GetObjectRequest object_request;
+  object_request.SetBucket(aws_bucket_name);
+  object_request.SetKey(aws_object_name);
+
+  auto get_object_outcome = s3_client.GetObject(object_request);
+  if (!get_object_outcome.IsSuccess()) {
+    auto error = get_object_outcome.GetError();
+    const std::string error_str = getErrorMessage(error);
+    throw std::runtime_error("DownloadNpyData error: " + error_str);
+  } else {
+    auto &image_file = get_object_outcome.GetResultWithOwnership().GetBody();
+    const char *output_filename = filename.c_str();
+    std::ofstream output_file(output_filename, std::ios::binary);
+    output_file << image_file.rdbuf();
+  }
+}
+
+template <typename T>
+int GetPreprocessNpyFile(string npy_filename,
+                         std::vector<unsigned long> &input_shape,
+                         std::vector<T> &input_data) {
+  bool fortran_order;
+  npy::LoadArrayFromNumpy(npy_filename, input_shape, fortran_order, input_data);
+  return 0;
+}
+
+void RunInference(const std::string &compiled_model,
+                  const std::string &npy_name) {
+  DLRModelHandle handle;
+
+  std::cout << "CreateDLRModel" << std::endl;
+  int dev_type = 1;  // cpu == 1
+  int dev_id = 0;
+  char *model_path = const_cast<char *>(compiled_model.c_str());
+  CreateDLRModel(&handle, model_path, dev_type, dev_id);
+
+  std::cout << "GetDLRNumOutputs" << std::endl;
+  int num_outputs;
+  GetDLRNumOutputs(&handle, &num_outputs);
+  std::vector<std::vector<float>> outputs;
+  for (int i = 0; i < num_outputs; i++) {
+    int64_t cur_size = 0;
+    int cur_dim = 0;
+    GetDLROutputSizeDim(&handle, i, &cur_size, &cur_dim);
+    std::vector<float> output(cur_size, 0);
+    outputs.push_back(output);
+  }
+
+  std::cout << "GetPreprocessNpyFile" << std::endl;
+  std::vector<unsigned long> in_shape_ul;
+  std::vector<float> input_data;
+  GetPreprocessNpyFile<float>(npy_name, in_shape_ul, input_data);
+  std::vector<int64_t> in_shape =
+      std::vector<int64_t>(in_shape_ul.begin(), in_shape_ul.end());
+
+  std::cout << "SetDLRInput" << std::endl;
+  string input_name = "data";
+  int64_t input_dimension = in_shape.size();
+  SetDLRInput(&handle, input_name.c_str(), in_shape.data(),
+              (float *)input_data.data(), static_cast<int>(input_dimension));
```

Comment:
cast to `(float *)` is not needed anymore. Input data param type was changed to `void*` recently